validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java

```diff
@@ -25,108 +25,135 @@
+    Map<String, Object> retrievedTrace = this.getRetrievedTrace(traceIdList);
+
+    // data model validation of other fields of segment document
+    for (Map.Entry<String, Object> entry : storedTrace.entrySet()) {
+      if (!entry.getValue().equals(retrievedTrace.get(entry.getKey()))) {
+        log.error("data model validation failed");
+        log.info("mis matched data model field list");
+        log.info("value of stored trace map: {}", entry.getValue());
+        log.info("value of retrieved map: {}", retrievedTrace.get(entry.getKey()));
+        log.info("==========================================");
+        throw new BaseException(ExceptionCode.DATA_MODEL_NOT_MATCHED);
+      }
     }
-
-    @Override
-    public void validate() throws Exception {
-        // get stored trace
-        Map<String, Object> storedTrace = this.getStoredTrace();
-
-        // create trace id list to retrieve trace from x-ray service
-        String traceId = (String) storedTrace.get("trace_id");
-        List<String> traceIdList = Collections.singletonList(traceId);
-
-        // get retrieved trace from x-ray service
-        Map<String, Object> retrievedTrace = this.getRetrievedTrace(traceIdList);
-
-        // validation of trace id
-        if (!storedTrace.get("trace_id").equals(retrievedTrace.get("trace_id"))) {
-            log.error("trace id validation failed");
-            throw new BaseException(ExceptionCode.TRACE_ID_NOT_MATCHED);
-        }
-
-        // data model validation of other fields of segment document
-        for (Map.Entry<String, Object> entry : storedTrace.entrySet()) {
-            if (!entry.getValue().equals(retrievedTrace.get(entry.getKey()))) {
-                log.error("data model validation failed");
-                log.info("mis matched data model field list");
-                log.info("value of stored trace map: {}", entry.getValue());
-                log.info("value of retrieved map: {}",retrievedTrace.get(entry.getKey()));
-                log.info("==========================================");
-                throw new BaseException(ExceptionCode.DATA_MODEL_NOT_MATCHED);
-            }
-        }
-    }
-
-    // this method will hit get trace from x-ray service and get retrieved trace
-    private Map<String, Object> getRetrievedTrace(List<String> traceIdList) throws Exception {
-        Map<String, Object> flattenedJsonMapForRetrievedTrace = null;
-        AtomicReference<List<Trace>> retrieveTraceListAtomicReference = new AtomicReference<>();
-        int MAX_RETRY_COUNT = 3;
-
-        RetryHelper.retry(MAX_RETRY_COUNT, () -> {
-            List<Trace> retrieveTraceList = null;
-            retrieveTraceList = xrayService.listTraceByIds(traceIdList);
-            retrieveTraceListAtomicReference.set(retrieveTraceList);
-
-            if (retrieveTraceList == null || retrieveTraceList.isEmpty()) {
-                throw new BaseException(ExceptionCode.EMPTY_LIST);
-            }
+  }
```

Comment:
Can we add a line here saying validator for this path is done successfully? Currently we don't see any kind of success log it would be good for debugging on Github actions.

Reply:
good catch!! let me add it